jaeger_client/span.py

```diff
@@ -78,12 +78,9 @@ def set_tag(self, key, value):
         :param value:
         """
         with self.update_lock:
-            if key == ext_tags.SAMPLING_PRIORITY:
-                if value > 0:
-                    self.context.flags |= SAMPLED_FLAG | DEBUG_FLAG
-                else:
-                    self.context.flags &= ~SAMPLED_FLAG
-            elif self.is_sampled():
+            if key == ext_tags.SAMPLING_PRIORITY and self._set_sampling_priority(value):
```

Comment:
if the priority flag setting was successful, shouldn't we add the tag? Here it looks like we skip on success.

Reply:
Wow good catch. I must have messed this up. I'm surprised it still seems to work in the crossdock test. Then again, the propagation there might act differently.